[Bug] When a newline is included in the Program Args of the application, it will cause a parsing error of Flink's ParameterTool class

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

When a newline is included in the Program Args of the application, it will cause a parsing error of Flink's ParameterTool class

### StreamPark Version

1.2.4

### Java Version

jdk8

### Flink Version

1.14.3

### Scala Version of Flink

2.12

### Error Exception

```log
Caused by: org.apache.flink.client.program.ProgramInvocationException: The main method caused an error: Error parsing arguments '[--a, b, --b, c, ]' on ''. Please prefix keys with -- or -.
	at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:372)
	at org.apache.flink.client.program.PackagedProgram.invokeInteractiveModeForExecution(PackagedProgram.java:222)
	at org.apache.flink.client.program.PackagedProgramUtils.getPipelineFromProgram(PackagedProgramUtils.java:158)
	at org.apache.flink.client.program.PackagedProgramUtils.createJobGraph(PackagedProgramUtils.java:82)
	at org.apache.streampark.flink.submit.trait.FlinkSubmitTrait$class.getJobGraph(FlinkSubmitTrait.scala:176)
	at org.apache.streampark.flink.submit.impl.YarnPerJobSubmit$.doSubmit(YarnPerJobSubmit.scala:83)
	at org.apache.streampark.flink.submit.trait.FlinkSubmitTrait$class.submit(FlinkSubmitTrait.scala:114)
	at org.apache.streampark.flink.submit.impl.YarnPerJobSubmit$.submit(YarnPerJobSubmit.scala:39)
	at org.apache.streampark.flink.submit.FlinkSubmit$.submit(FlinkSubmit.scala:32)
	at org.apache.streampark.flink.submit.FlinkSubmit.submit(FlinkSubmit.scala)
```


### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala
+++ b/streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala
@@ -361,55 +361,52 @@ trait FlinkSubmitTrait extends Logger {
     val programArgs = new ArrayBuffer[String]()
 
     if (StringUtils.isNotEmpty(submitRequest.args)) {
+      val multiLineChar = "\"\"\""
+      val array = submitRequest.args.split("\\s+")
+      if (array.filter(_.startsWith(multiLineChar)).isEmpty) {
+        array.foreach(programArgs +=)
+      } else {
+        val argsArray = new ArrayBuffer[String]()
+        val tempBuffer = new ArrayBuffer[String]()
 
-      val array = submitRequest.args.split("\\s")
-      val argsArray = new ArrayBuffer[String]()
-      val tempBuffer = new ArrayBuffer[String]()
-
-      def processElement(index: Int, num: Int): Unit = {
-
-        if (index == array.length) {
-          if (tempBuffer.nonEmpty) {
-            argsArray += tempBuffer.mkString(" ")
+        def processElement(index: Int, multiLine: Boolean): Unit = {
+          if (index == array.length) {
+            if (tempBuffer.nonEmpty) {
+              argsArray += tempBuffer.mkString(" ")
+            }
+            return
           }
-          return
-        }
+          val next = index + 1
+          val elem = array(index)
 
-        val next = index + 1
-        val elem = array(index)
-
-        if (elem.trim.nonEmpty) {
-          if (num == 0) {
-            if (elem.startsWith("'")) {
-              tempBuffer += elem
-              processElement(next, 1)
-            } else if (elem.startsWith("\"")) {
-              tempBuffer += elem
-              processElement(next, 2)
+          if (elem.trim.nonEmpty) {
+            if (!multiLine) {
+              if (elem.startsWith(multiLineChar)) {
+                tempBuffer += elem.drop(3)
+                processElement(next, true)
+              } else {
+                argsArray += elem
+                processElement(next, false)
+              }
             } else {
-              argsArray += elem
-              processElement(next, 0)
+              if (elem.endsWith(multiLineChar)) {
+                tempBuffer += elem.dropRight(3)
+                argsArray += tempBuffer.mkString(" ")
+                tempBuffer.clear()
+                processElement(next, false)
+              } else {
+                tempBuffer += elem
+                processElement(next, multiLine)
+              }
             }
           } else {
             tempBuffer += elem
-            val end1 = elem.endsWith("'") && num == 1
-            val end2 = elem.endsWith("\"") && num == 2
-            if (end1 || end2) {
-              argsArray += tempBuffer.mkString(" ")
-              tempBuffer.clear()
-              processElement(next, 0)
-            } else {
-              processElement(next, num)
-            }
+            processElement(next, false)
           }
-        } else {
-          tempBuffer += elem
-          processElement(next, 0)
         }
+        processElement(0, false)
+        argsArray.foreach(x => programArgs += x.trim)
       }
-
-      processElement(0, 0)
-      argsArray.foreach(x => programArgs += x.trim.replaceAll("^[\"|']|[\"|']$", ""))
     }
 
     if (submitRequest.applicationType == ApplicationType.STREAMPARK_FLINK) {

--- a/streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/test/scala/org/apache/streampark/flink/submit/test/ParameterTestCase.scala
+++ b/streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/test/scala/org/apache/streampark/flink/submit/test/ParameterTestCase.scala
@@ -16,6 +16,7 @@
  */
 package org.apache.streampark.flink.submit.test
 
+import org.apache.commons.lang3.StringUtils
 import org.apache.flink.api.java.utils.ParameterTool
 import org.junit.jupiter.api.{Assertions, Test}
 
@@ -44,68 +45,66 @@ class ParameterTestCase {
   }
 
   @Test def testExtractProgramArgs(): Unit = {
-
-    val argsStr = "--url localhost:8123 \n" +
-      "--insertSql1 'insert \'\'into default.test values (?,?,?,?,?)' \n" +
-      "--insertSql2 'insert into default.test values (1,2,3,4,\"111\")'\n " +
-      "--insertSql2 \"insert into default.test values (1,2,3,4,\'111\')\" \n" +
-      "--insertSql2 'insert into default.test values (1,2,3,4,\"111\", \'22\', \'\')'"
-
-    val array = argsStr.split("\\s")
-    val argsArray = new ArrayBuffer[String]()
-    val tempBuffer = new ArrayBuffer[String]()
-
-    def processElement(index: Int, num: Int): Unit = {
-
-      if (index == array.length) {
-        if (tempBuffer.nonEmpty) {
-          argsArray += tempBuffer.mkString(" ")
-        }
-        return
-      }
-
-      val next = index + 1
-      val elem = array(index)
-
-      if (elem.trim.nonEmpty) {
-        if (num == 0) {
-          if (elem.startsWith("'")) {
-            tempBuffer += elem
-            processElement(next, 1)
-          } else if (elem.startsWith("\"")) {
-            tempBuffer += elem
-            processElement(next, 2)
-          } else {
-            argsArray += elem
-            processElement(next, 0)
-          }
-        } else {
-          tempBuffer += elem
-          val end1 = elem.endsWith("'") && num == 1
-          val end2 = elem.endsWith("\"") && num == 2
-          if (end1 || end2) {
-            argsArray += tempBuffer.mkString(" ")
-            tempBuffer.clear()
-            processElement(next, 0)
-          } else {
-            processElement(next, num)
-          }
-        }
+    val argsStr = "--host localhost:8123\n\n\n" +
+      "--sql \"\"\"insert into table_a select * from table_b\"\"\"\n" +
+      "--c d\r\n" +
+      "--x yyy"
+    val programArgs = new ArrayBuffer[String]()
+    if (StringUtils.isNotEmpty(argsStr)) {
+      val multiLineChar = "\"\"\""
+      val array = argsStr.split("\\s+")
+      if (array.filter(_.startsWith(multiLineChar)).isEmpty) {
+        array.foreach(programArgs +=)
       } else {
-        tempBuffer += elem
-        processElement(next, 0)
+        val argsArray = new ArrayBuffer[String]()
+        val tempBuffer = new ArrayBuffer[String]()
+
+        def processElement(index: Int, multiLine: Boolean): Unit = {
+
+          if (index == array.length) {
+            if (tempBuffer.nonEmpty) {
+              argsArray += tempBuffer.mkString(" ")
+            }
+            return
+          }
+
+          val next = index + 1
+          val elem = array(index)
+
+          if (elem.trim.nonEmpty) {
+            if (!multiLine) {
+              if (elem.startsWith(multiLineChar)) {
+                tempBuffer += elem.drop(3)
+                processElement(next, true)
+              } else {
+                argsArray += elem
+                processElement(next, false)
+              }
+            } else {
+              if (elem.endsWith(multiLineChar)) {
+                tempBuffer += elem.dropRight(3)
+                argsArray += tempBuffer.mkString(" ")
+                tempBuffer.clear()
+                processElement(next, false)
+              } else {
+                tempBuffer += elem
+                processElement(next, multiLine)
+              }
+            }
+          } else {
+            tempBuffer += elem
+            processElement(next, false)
+          }
+        }
+
+        processElement(0, false)
+        argsArray.foreach(x => programArgs += x.trim)
       }
     }
 
-    processElement(0, 0)
-
-    val programArgs = argsArray.map(_.trim.replaceAll("^[\"|']|[\"|']$", "")).toList
-
     Assertions.assertEquals("localhost:8123", programArgs(1))
-    Assertions.assertEquals("insert \'\'into default.test values (?,?,?,?,?)", programArgs(3))
-    Assertions.assertEquals("insert into default.test values (1,2,3,4,\"111\")", programArgs(5))
-    Assertions.assertEquals("insert into default.test values (1,2,3,4,\'111\')", programArgs(7))
-    Assertions.assertEquals("insert into default.test values (1,2,3,4,\"111\", \'22\', \'\')", programArgs(9))
-
+    Assertions.assertEquals("insert into table_a select * from table_b", programArgs(3))
+    Assertions.assertEquals("d", programArgs(5))
+    Assertions.assertEquals("yyy", programArgs(7))
   }
 }

